### PR TITLE
completed test_da_dp_tc069_tc070_tc071_tc072_tc073

### PR DESCRIPTION
**Pull request by vophi027**
[report.zip](https://github.com/vantruongt2/PySelenium/files/3161546/report.zip)

### CODE REVIEW

File: tests_phi/pages/dashboard_page.py [hunk 1]
```diff
@@ -82,3 +83,10 @@ def delete_page(self, page_name):
         self.select_global_selecting_item("Delete")
         browser.get_alert().accept()
```
Comment (trungtranjj) [inline tests_phi/pages/dashboard_page.py]: You have method browser_helper, You should use from it

File: tests_phi/pages/general_setting_pages.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    def __init__(self):
+        self.dynamic_button = BaseElement("//input[@value='%s']")
+        self._txt_name = TextBox("id=txtProfileName")
+        self._cbb_type = ComboBox("id=cbbEntityType")
+        self._cbb_related_data = ComboBox("id=cbbSubReport")
+        
+    def enter_general_setting_info(self, name, item_type = None, related_data = None):
+        self._txt_name.wait_for_visible()
+        self._txt_name.find_element().clear()
```
Comment (trungtranjj) [inline tests_phi/pages/general_setting_pages.py]: should write clear() in element/textbox method

File: tests_phi/testcases/data_profiles.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    @pytest.mark.usefixtures("clean_up") 
+    def test_da_dp_tc072_t073(self):
+        ''' - Verify that all data profile types are listed under "Item Type" dropped down menu
+            - Verify that all data profile types are listed in priority order under "Item Type" dropped down menu
+        '''
+        time_types_list = ["Test Modules", "Test Cases",
+                            "Test Objectives", "Data Sets",
+                             "Actions", "Interface Entities",
+                             "Test Results", "Test Case Results",
+                              "Test Suites","Bugs"]
```
Comment (trungtranjj) [inline tests_phi/testcases/data_profiles.py:67]: Should put test data outside the test method

File: tests_phi/pages/general_setting_pages.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+class DAGeneralSettingsPage():
+
+    def __init__(self):
+        self.dynamic_button = BaseElement("//input[@value='%s']")
+        self._txt_name = TextBox("id=txtProfileName")
+        self._cbb_type = ComboBox("id=cbbEntityType")
+        self._cbb_related_data = ComboBox("id=cbbSubReport")
+        
+    def enter_general_setting_info(self, name, item_type = None, related_data = None):
```
Comment (vantruongt2) [inline tests_phi/pages/general_setting_pages.py]: the name should be fill instead of enter

File: tests_phi/testcases/data_profiles.py (hunk 3 above)
Comment (vophi027) [reply tests_phi/testcases/data_profiles.py:67]: It is used in a test case only

**Review by vantruongt2 — CHANGES_REQUESTED** [review]

File: tests_phi/pages/general_setting_pages.py (hunk 2 above)
Comment (vophi027) [reply tests_phi/pages/general_setting_pages.py]: done

File: tests_phi/pages/dashboard_page.py (hunk 1 above)
Comment (vophi027) [reply tests_phi/pages/dashboard_page.py]: done

File: tests_phi/testcases/data_profiles.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    
+    @pytest.mark.usefixtures("clean_up")   
+    @pytest.mark.parametrize("profile_name,err_message", data_test)
+    def test_da_dp_tc069_tc070_tc071(self, profile_name, err_message):
+        ''' - Verify that user is unable to proceed to next step or finish creating data profile if  "Name *" field is left empty
+            - Verify that special characters ' /:*?<>|"#[]{}=%; 'is not allowed for input to "Name *" field
+            - Verify that Data Profile names are not case sensitive
+        '''     
+        logging.info("1. Log in Dashboard")
+        self.login_page.go_to()
```
Comment (vantruongt2) [inline tests_phi/testcases/data_profiles.py]: Can you make the test run faster? Its good if we don't need to login/logout on each test data

**Review by vantruongt2 — CHANGES_REQUESTED** [review]

File: tests_phi/pages/general_setting_pages.py (hunk 4 above)
Comment (vophi027) [reply tests_phi/pages/general_setting_pages.py]: updated

File: tests_phi/testcases/data_profiles.py (hunk 5 above)
Comment (vophi027) [reply tests_phi/testcases/data_profiles.py]: Optimized this suite to reduce running time

**Pull request merged**